src/components/ui/PuzzleModal.tsx

```
@@ -0,0 +1,638 @@
+        </AnimatePresence>
+
+        {/* Sentient AI Modal */}
+        <SentientAIModal
+          isOpen={showSentientAI}
+          difficulty={puzzle.difficulty}
+          puzzleType={puzzle.type}
+          answer={Array.isArray(puzzle.answer) ? puzzle.answer[0] : puzzle.answer}
+          onClose={() => {
+            setShowSentientAI(false);
+            setLifelinesUsed(prev => prev + 1);
+            lifelineManager.useSentientAI(puzzle.id);
```

Biome (2.1.2)

[error] This hook is being called from a nested function, but all hooks must be called unconditionally from the top-level component.

For React to preserve state between calls, hooks needs to be called unconditionally and always in the same order.
See https://reactjs.org/docs/hooks-rules.html#only-call-hooks-at-the-top-level

(lint/correctness/useHookAtTopLevel)